### ISSUE TITLE
Registry tool: Kubernetes-style resource import

### ISSUE THREAD

**Issue by timburks** [apigee/registry#450]
We've prototyped a variety of ways to use the Registry API to add information to the registry, including custom importers for the Google API Discovery Format, the public Protocol Buffer descriptions of Google APIs, and the apis.guru OpenAPI Directory. But these are all very specialized and seem unlikely to be useful by teams in practice.

We could leave it to users to write their own importers, but we've noticed that other projects have been following a pattern that we first saw in Kubernetes - describing [objects](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/) (resources) with YAML that is in a standardized wrapper that includes API version and a type identifier, along with other metadata, to identify the type of object being described. This allows users to create or modify objects using a single command: [`kubectl apply`](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#apply).

Closely following the Kubernetes pattern, a YAML file for a sample API might look like this:
```
apiVersion: registry/v1
kind: API
metadata:
  name: registry
data:
  displayName: "Apigee Registry API"
  description: "The Registry API allows teams to track and manage machine-readable descriptions of APIs."
  availability: PREVIEW
  recommendedVersion: v1
  recommendedDeployment: 
  versions:
  - metadata:
      name: v1
    data:
      displayName: "v1"
      description: "New in 2022"
      state: STAGING
      specs:
      - metadata:
          name: "protos"
        data:
          filename:
          description:
          sourceURI:
  deployments:
  - metadata:
      name: prod
    data:
      displayName: Production
      description:
      specRevision:
      endpointURI: https://apigeeregistry.googleapis.com
      externalChannelURI:
      intendedAudience:
      accessGuidance:
```

Analogously to `kubectl`, the `registry` tool could have an `apply` subcommand and if we followed the `kubectl` pattern for resource access,  `registry get <resourcename> -o yaml` would return a YAML representation of objects in a form consistent with the example above.

**Comment by timburks** [comment]
Here's an [example from Backstage](https://github.com/backstage/backstage/blob/master/packages/catalog-model/examples/components/petstore-component.yaml) that shows the use of this pattern to implement the backing store for a web site in a Git repository.

/cc @theganyo for other possible examples

**Comment by timburks** [comment]
Work in progress in https://github.com/timburks/registry/tree/patch-yaml

**Comment by shrutiparabgoogle** [comment]
Does the yaml format (specially the information in `spec` field), has any connection with the protos that we already have? Else, we will have to maintain definitions at two different places `protos` and `yaml`, and these can easily diverge from one another.

**Comment by timburks** [comment]
@shrutiparabgoogle That's true. It's not the first redundancy - we also have other places where we have to maintain duplicative definitions (the storage models, for example). I think at our current scale, it would be ok, and later if needed we could figure out a way to generate these automatically. You might think of this as an effort to work backwards from a familiar representation (Kubernetes-style objects) to the Registry API.

**Comment by timburks** [comment]
Here's a directory of generated YAML files representing APIs: https://github.com/timburks/registry-experimental/tree/exports/exports/googleapis/apis

**Comment by timburks** [comment]
Other related Kubernetes links:
- [`kubectl create`](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#create) creates resources (an apparent subset of `kubectl apply`).
- ['kubectl get'](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#get) gets resources, optionally in YAML format.

**Comment by timburks** [comment]
Addressed in #463
